### PR TITLE
AO3-5535 Remove series index

### PR DESCRIPTION
**Pull request by Bilka2**
# Pull Request Checklist

* [X] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [X] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [X] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [X] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [X] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-5535

## Purpose

Remove series index, so accessing `/series` redirects to the homepage and gives a red flash error that asks, "Whose series did you want to see?"

## Notes

As visible in the tests, trying to access the series of a user that does not exist like `/users/invalid_user/series` also redirects.
Previously, this page would throw a `ActiveRecord::RecordNotFound` error and 404. The behaviour I implemented here mirrors that of gift_controller, which also redirects for an invalid user name.

Trying to access the series of an invalid pseud (`users/valid_user/pseuds/invalid_pseud/series`) does not redirect. I left it with the original behaviour because I don't know if it should be changed, since gift_controller doesn't handle pseuds. However, I think it would make sense if this also redirected.

## References

I referenced the gift_controller for code and behaviour.

## Credit

Bilka (he/him)

### CODE REVIEW

File: app/controllers/series_controller.rb [hunk 1]
```diff
@@ -16,21 +16,20 @@ def load_series
...
+      redirect_to(root_path) and return
+    end
+
+    @page_subtitle = ts("%{username} - Series", username: @user.login)
+    pseuds = @user.pseuds
+    if params[:pseud_id]
+      @pseud = @user.pseuds.find_by(name: params[:pseud_id])
+      raise ActiveRecord::RecordNotFound, "Couldn't find pseud '#{params[:pseud_id]}'" unless @pseud
+      
+      @page_subtitle = ts("by ") + @pseud.byline
```
Comment (Bilka2) [inline app/controllers/series_controller.rb:31]: This page_subtitle is only "by pseud (user)", it should probably include the word "series" somewhere, but that is outside the scope of this PR. Is the [support form](https://archiveofourown.org/support) the right place to report issues like this?

Comment (sarken) [reply app/controllers/series_controller.rb:31]: Here is fine! I've made [an issue](https://otwarchive.atlassian.net/browse/AO3-6378).

**Review by sarken — CHANGES_REQUESTED** [review]
Thanks for bearing with us! I had to do some research to find out what our actual desired behavior was for instances where the user or pseud wasn't found. While we do want to redirect when no user params are provided, we'd like to give 404 errors when the user or pseud is nonexistent -- that gives the person who entered the URL a chance to fix any typos. 

(The gifts behavior is incorrect and I'll have to make an issue for updating that.)

**Review by sarken — APPROVED** [review]
Thank you!
